manage-data/ingest/ingesting-data-from-applications/ingest-logs-from-nodejs-web-application-using-filebeat.md

```diff
@@ -20,514 +22,443 @@ This guide demonstrates how to ingest logs from a Node.js web application and de
+    } catch (err) {
+      console.log("cannot execute request:", err)
+    }
+    await sleep(Math.random() * 5000 + 2000)
+  }
+})()
+```
 
-    ```json
-    {"@timestamp":"2021-09-09T18:42:20.799Z","log.level":"info","message":"handled request","ecs":{"version":"1.6.0"},"http":{"version":"1.1","request":{"method":"POST","headers":{"user-agent":"got (https://github.com/sindresorhus/got)","from":"octopus@the.zoo","accept":"application/json","accept-encoding":"gzip, deflate, br","host":"localhost:3000","connection":"close","content-length":"0"},"body":{"bytes":0}},"response":{"status_code":200,"headers":{"foo":"Bar"}}},"url":{"path":"/","full":"http://localhost:3000/"},"client":{"address":"::ffff:127.0.0.1","ip":"::ffff:127.0.0.1","port":49930},"user_agent":{"original":"got (https://github.com/sindresorhus/got)"}}
-    ```
+This Node.js app generates HTTP requests with a random method of type `GET`, `POST`, or `PUT`, and a random `from` request header using various pretend email addresses. The requests are sent at random intervals between 2 and 7 seconds.
 
-    Each log entry contains details of the HTTP request. In particular, in this example you can find the timestamp of the request, a request method of type `PUT`, and a request `from` header with the email address `octopus@the.zoo`. Your example will likely be a bit different since the request type and the email address are generated randomly.
+The [Got package](https://www.npmjs.com/package/got) is used to send the requests, and they are directed to your web server at `http://localhost:3000`. To learn about sending custom headers such as the `from` field used in this example, check [headers](https://github.com/sindresorhus/got/blob/0fb6ec60d299fd9b48966608a4c3f201746d821c/documentation/2-options.md#headers) in the Got documentation.
 
-    Having your logs written in a JSON format with ECS fields allows for easy parsing and analysis, and for standardization with other applications. A standard, easily parsible format becomes increasingly important as the volume and type of data captured in your logs expands over time.
+::::
 
-4. After confirming that both `webserver.js` and `webrequests.js` run as expected, enter *CTRL + C* to stop the Node.js script, and also delete `log.json`.
+::::{step} Run the traffic generator
 
+In a new terminal window, give the Node.js script a trial run:
 
-## Set up Filebeat [ec-node-logs-filebeat]
+```bash
+node webrequests.js
+```
 
-Filebeat offers a straightforward, easy to configure way to monitor your Node.js log files and port the log data into your deployment.
+::::
 
-**Get Filebeat**
+::::{step} Verify log generation
 
-[Download Filebeat](https://www.elastic.co/downloads/beats/filebeat) and unpack it on the local server from which you want to collect data.
+After the script has run for about 30 seconds, enter *CTRL + C* to stop it. Have a look at your Node.js `logs/log.json` file. It should contain some entries like this one:
```

Comment:
If you want to try it out, you could consider using [docs-builder's keyboard shortcut syntax](https://elastic.github.io/docs-builder/syntax/kbd/) instead of `CTRL + C`, but it's up to you since we definitely aren't using that consistently across the docs right now.